bugtool/configuration/configuration.go

```diff
@@ -0,0 +1,371 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package configuration
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/cilium/cilium/bugtool/dump"
+	"github.com/cilium/cilium/bugtool/options"
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/mountinfo"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func deduplicate(arr []string) []string {
+	sort.Strings(arr)
+	if len(arr) <= 1 {
+		return arr
+	}
+	narr := []string{arr[0]}
+	for i := 1; i < len(arr); i++ {
+		if arr[i] != arr[i-1] {
+			narr = append(narr, arr[i])
+		}
+	}
+	return narr
+}
+
+// CreateDump creates the root dump task to be used for bugtool dumps.
+// If a configuration file is specified, it will attempt to use that
+// to load a task, otherwise it uses the "topics" list to compose a
+// a task.
+func CreateDump(conf *options.Config) dump.Task {
+	if conf.ConfigFile != "" && !(conf.Generate || conf.DryRun) {
+		log.Infof("Using specified config file: %s", conf.ConfigFile)
+		fd, err := os.Open(conf.ConfigFile)
+		if err != nil {
+			log.Fatalf("Failed to open config file %s: %v", conf.ConfigFile, err)
+		}
+		t, err := dump.Decode(fd)
+		if err != nil {
+			log.Fatalf("Failed to decode config %s: %v", conf.ConfigFile, err)
+		}
+		return t
+	}
+
+	// Handle deprecated topic specific flags.
+	if conf.EnvoyDump {
+		log.Info("[Deprecated] Use --topics=envoy instead")
+		conf.Topics = append(conf.Topics, "envoy")
+	}
+
+	if conf.GetPProf {
+		log.Info("[Deprecated] Use --topics=pprof instead")
+		conf.Topics = append(conf.Topics, "pprof")
+	}
+
+	ts := dump.Tasks{}
+	conf.Topics = deduplicate(conf.Topics)
+	log.Debugf("Final topics list %v", conf.Topics)
+	logs, unstructured, structured := defaultResources()
+	for _, topic := range conf.Topics {
```

Comment:
Could we drop the `topics` list and just have a config yaml defined somewhere in this package that has default tasks? Or do we do anything programmatically that isn't possible via yaml configuration?

This yaml could be helpful as documentation for how to use the config files, and we could drop all the code that goes over topics since we already have whole `Decode` package that could generate tasks from hardcoded yaml.

Reply:
> Or do we do anything programmatically that isn't possible via yaml configuration?

No, the topics is just a convenient shorthand for just running a logical subset of tasks.

My reasoning for it is:
* Its a clean way to replace the --get-pprof and --envoy-dump flags. i.e. --topics=envoy.
* It'll be easy/clean for tools consuming this to just specify a topic, instead of having to generate/copy config around.